packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,8 +115,15 @@ const withInfiniteScroll = (
       };
 
       this.props.dataSource.onDataChange(this.updateData);
+
+      const thresholds = {
+        small: 0.01,
+        half: 0.5,
+        full: 0.99, // using 0.99 instead of 1 to avoid problems with float precision in IE11
+      };
+      const displaySize = this.props.loaderMinDisplay || 'full';
       this.observer = new IntersectionObserver(this.handleIntersection, {
-        threshold: 0.99, // using 0.99 instead of 1 to avoid problems with float precision in IE11
+        threshold: thresholds[displaySize],
```

Comment:
It's worth adding a default here, the prop type will only trigger a warning if the wrong value is passed and in that case the component will break. Can be something like `thresholds[displaySize] || thresholds .full`.

Reply:
true after several changes I missed the case where the prop is not empty but invalid